lib/consyncful/item_mapper.rb

```diff
@@ -5,12 +5,16 @@ module Consyncful
   # Responsible for mapping an update received from Contentful's syncronisation API
   # into useful fields for Consyncful::PersistedItem to store in the database.
   class ItemMapper
+    attr_reader :item
+
     def initialize(item)
       @item = item
     end
 
     def deletion?
-      @item.is_a?(Contentful::DeletedEntry) || @item.is_a?(Contentful::DeletedAsset)
+      @item.is_a?(Contentful::DeletedEntry) || 
+        @item.is_a?(Contentful::DeletedAsset) ||
+        tagged_content?(@item)
     end
```

Comment:
I would you prefer you to move the tag logic to another method here with a descriptive name. While it's effect is the same, removing and item because it's tag doesn't match is different from the sync API's deletion messages.

```ruby 
def excluded_by_tag?
  tagged_content?(@item)
end
```

and then in PersistedItem, 

```ruby
if @item.deletion? || @item.excluded_by_tag?
  delete_model(@item.id, @stats)
```